framework/validators/UrlValidator.php

```diff
@@ -28,7 +28,7 @@ class UrlValidator extends Validator
      * The pattern may contain a `{schemes}` token that will be replaced
      * by a regular expression which represents the [[validSchemes]].
      */
-    public $pattern = '/^{schemes}:\/\/(([A-Z0-9][A-Z0-9_-]*)(\.[A-Z0-9][A-Z0-9_-]*)+)/i';
+    public $pattern = '/^({schemes}:\\/\\/)(www\\.)?([0-9u0000-uFFFF]*\\.\\-)?([0-9u0000-uFFFF\\-]*)\\.([a-zA-Z]{2,6})?([u0000-uFFFF]{2,6})\\/?([u0000-uFFFF0-9_\\-\\?=\\.\\&])*?/i';
```

Comment:
> I added www as URL might have www in their name. It is optional, I can
> remove that, just thought that would be good addition.

The regular expression MUST allow `www.domain.com` as a regular case, 'cause it's the same as `subdomain.domain.com`